_posts/2023-02-01-fooconf-helsinki.md

```diff
@@ -0,0 +1,91 @@
+- Share what you learned!
+- Do good
+- Be yourself
+
+The speech was really interesting and provided something to think about. The topics covered were common to many developers.
+
+![paste](/img/2023-fooconf/fooconf2023_4.jpg)
+_Learning by Tinkering By Tom Cools_
+
+After the first speech, we took some different tracks. Teemu listened to Cloud-native dev tools by **Grace Jansen** from IBM. It was a good introduction to OSS Java cloud-native technologies like Open Liberty, MicroProfile, Jakarta EE, testing (TestContainers, Microshed) and deploying (Buildpacks.io, Paketo buildpacks, Telepresence). It was a good overview and something to check for future projects as well. Nice demos, tips and tricks. Good to check slides for more information and links.
+
+Markus went to listen to the speech about JavaScript frameworks of tomorrow by **Juho Vepsäläinen** of Aalto University. The speech presented first the past and current trends in frameworks but quickly went on to prospects and the idea of Transitional Web Applications (TWAs). The main goal is to minimize the amount of loaded JavaScript and increase the amount of static content on pages, leading to more efficient web applications. Two approaches of these were introduced: the Island Architecture (11ty/is-land, îles, Capri and Astro -frameworks) and Resumability (Qwik-framework), and demos for Astro and Qwik were shown. Juho also presented his Ph.D. project called Gustwind (with side projects), which is using JSON as the building block for web applications.
+
+Next in line for Teemu was a very nice speech about how a Legacy app went to serverless bar by **Sébastien Blanc** from Aiven. It was a humorous story featuring a 13 years old legacy app called Bob which wanted to go cloud and serverless. The story went through very humorously including funny characters like Join the table, Maria, Kafka river etc and ended up in an Aiven product demo, about serverless, Kafka, Kubernetes and autoscaling.
+
+![paste](/img/2023-fooconf/fooconf2023_5.jpg)
+_Legacy app went to serverless bar by Sébastien Blanc_
+
+Let’s do a thing and call it Foo by **Maaret Pyhäjärvi** was next in line for Markus. It was a speech about how to approach testing, which was defined as “Find (some of) what others may have missed”. The main takeaways from the speech were: testing is the responsibility of the whole team (even a team with designated testers), KNOW the domain where the application will be used (e.g. in clocks number 4 in roman numerals is IIII not IV) and choosing the right level of test coverage. Finally, the speech gave some useful insights from Maaret's own experiences in testing.
+
+After lunch, we both listened to the speech about Security by design by **Daniel Deogun** from OmegaPoint. It was about how to implicitly write safe software by going through the log4j vulnerabilities as an example. We learned that good design mitigates risks but not fully. Good discussion about what is safe and what is not.
```

Comment:
*There was good discussion about what is safe and what is not.

^ would make for a complete sentence with a verb

Reply:
Fixed, thanks!